src/script_lua.c

```diff
@@ -1419,9 +1541,34 @@ void luaCallFunction(scriptRunCtx* run_ctx, lua_State *lua, robj** keys, size_t
     }
 
     if (err) {
-        addReplyErrorFormat(c,"Error running script (call to %s): %s\n",
-                run_ctx->funcname, lua_tostring(lua,-1));
-        lua_pop(lua,1); /* Consume the Lua reply and remove error handler. */
+        /* Error object is a table of the following format:
+         * {err='<error msg>', err_code='<error code>', source='<source file>', line=<line>}
+         * We can constract the error message from this inforamtion */
+        if (!lua_istable(lua, -1)) {
+            /* Should not happened, and we should considered assert it */
+            addReplyErrorFormat(c,"Error running script (call to %s)\n", run_ctx->funcname);
+        } else {
+            errorInfo err_info = {0};
+            sds final_msg = sdsempty();
+            luaExtractErrorInformation(lua, &err_info);
+            final_msg = sdscatfmt(final_msg, "-%s %s",
+                                  err_info.error_code,
+                                  err_info.msg);
+            if (err_info.line && err_info.source) {
+                final_msg = sdscatfmt(final_msg, " script: %s, on %s:%s.",
+                                      run_ctx->funcname,
+                                      err_info.source,
+                                      err_info.line);
+            }
+            addReplyErrorSdsExt(c, final_msg, err_info.ignore_err_stats_update? ERR_REPLY_FLAG_NO_STATS_UPDATE : 0);
+            if (err_info.ignore_err_stats_update) {
+                /* If we do not update the error stats, the current command stats will not be updated.
+                 * Update it manually here */
+                c->cmd->failed_calls++;
```

Comment:
this is a little bit odd, i suppose we have no other way (since we don't wanna increment `total_error_replies` and then we can't cause `call` to increment `failed_calls` for us.
i see two way to slightly improve (we can probably take both).
1. add a bit more words about the hack, i.e. the fact that we want increment failed_calls, to match the increment in `calls` later in `call()`, but we don't wanna increment `total_error_replies`. mention this is a dirty hack).
2. we can pass this increment to be done inside `afterErrorReply`, so this hack is in a more central location, and not hidden in script_lua.c.

Reply:
Fixed